docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+In a leader-based consensus, the network can (in the best case) progress if the validators turn their nodes exactly two times (i.e., two times of broadcasts). However, in a leaderless consensus, the number of broadcasts is proportional to the number of validators. This will never be acceptable in our essential assumption of light-weight node operation.
+
+TODO: check it and add reference
+
+### Veto the Round
+
+Again it can't be handled by the consensus layer because neither laziness nor censorship is a violation of the consensus protocol. However, we can find a detour if we **exploit the network asynchrony**. If a validator concludes that the current round should be escaped because of the honest-but-lazy-or-bad leader, they can just **pretend that the round is timed out**. This provides a very special feature over plain Tendermint; **veto the leader**.
+
+Since Tendermint is based on rounds, there is a special mechanism called *nil-vote*. Nodes can cast a nil-vote after the timeout, to escape the current round. This works similarly to the non-nil case, in that it takes two steps.
+
+In plain Tendermint, if the validators use nil-votes to veto the proposer, it just works because it's not so different from a rash timeout, but it might get stuck until the timeout in the split case. (Think about the case of 6/10 non-nil-votes and 3/10 nil-votes, 1/10 Unresponsive Byzantine nodes)
+Again, because we're using very long rounds, it will waste a lot of time, waiting for the timeout of the round in that kind of situation. If we encounter this unlucky case of a split, it is no better than the governance's solution (no approval of agendas).
+
+### Vetomint
+
+To improve this, we propose a variation of Tendermint, called *Vetomint*, which is effective in the 'split' case.
+
+### Properties
+
+1. `<1/6` BFT
+2. It is allowed (i.e., considered as a non-byzantine behavior) to cast a nil-vote before the timeout, with *some kind of reason*.
+3. If all the honest nodes cast either a vote or nil-vote, it is guaranteed that the round progresses; the **early termination**.
+4. If all the honest nodes cast a vote, it is guaranteed that the block progresses.
+5. If over 2/3 of the nodes cast a vote, the block progresses with the probability of `>0`.
+
+### How does it work?
+
+In Vetomint, unlike Tendermint, the round **always instantly progresses** if over 5/6 nodes either non-nil-vote or nil-vote. Especially when 5/6 nodes vote, it is guaranteed that the height progresses. The threshold 5/6 is just enough to ensure that all nodes observe at least 2/3 (Tendermint majority) of non-nil-votes no matter the individual arrival order of the votes, even with the possible 1/6 nil-votes.
+Note that the 'arrival order' matters, unlike Tendermint, because we have a new 'early termination' condition.
+
+Details are explained in issue #4 and the pseudo-code is [here](./vetomint-spec.pdf)
```

Comment:
Add a link to the issue.
https://github.com/postech-dao/simperby/issues/4

Reply:
updated